src/test/database/remove.spec.js

```diff
@@ -0,0 +1,194 @@
+"use strict";
+
+const chai = require("chai");
+const expect = chai.expect;
+const pathLib = require("path");
+const DatabaseRemove = require("../../database/remove");
+
+class TestDatabaseRemoveHelper {
```

Comment:
This is... interesting that it needs to exist. I get that you can provide your own helper, but you could require `DatabaseRemoveHelper` and use [sinon](https://sinonjs.org/releases/v7.1.0/) to stub out behavior you're interested in.

The fact that there's tests for the `TestDatabaseRemoveHelper` shows me that this is rather complex logic - using `sinon` can specify the exact behavior you want on the real `DatabaseRemoveHelper` (instead of passing in a custom one) - though a custom one should be able to be provided (and should be called).

I hope that makes sense...? Let me know if it doesn't.

Reply:
Yeah, I do agree `TestDatabaseRemoveHelper` is quite complicated, because RTDB does not allow path with no leaf.

I find it quite tricky to mock them up, because the `DatabaseRemove` is expected to call same shallow get method multiple times. I am not sure how to mock this mutating behavior. 
Also, throttled request queue makes the order of the request undefined, which is expected, but that makes it even hard to mock the API in a way that is not flaky.